flutter/tonic/dart_isolate_reloader.cc

```diff
@@ -0,0 +1,327 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/tonic/dart_isolate_reloader.h"
+
+#include "base/bind.h"
+#include "base/callback.h"
+#include "base/threading/thread.h"
+#include "flutter/tonic/dart_api_scope.h"
+#include "flutter/tonic/dart_converter.h"
+#include "flutter/tonic/dart_dependency_catcher.h"
+#include "flutter/tonic/dart_error.h"
+#include "flutter/tonic/dart_isolate_scope.h"
+#include "flutter/tonic/dart_library_provider.h"
+#include "flutter/tonic/dart_library_loader.h"
+#include "flutter/tonic/dart_state.h"
+#include "flutter/tonic/monitor.h"
+#include "mojo/data_pipe_utils/data_pipe_drainer.h"
+
+using mojo::common::DataPipeDrainer;
+
+namespace blink {
+
+// As each source file is loaded, a LoadResult is queued to be processed on the
+// isolate's thread. A LoadResult contains the payload or an error message.
+class DartIsolateReloader::LoadResult {
+ public:
+  // Successful load result.
+  LoadResult(intptr_t tag,
+             const std::string& url,
+             const std::string& library_url,
+             const std::vector<uint8_t>& payload)
```

Comment:
This should be `std::vector<uint8_t>` rather than `const std::vector<uint8_t>&` so that the `std::move` below works.  What you have currently is still a memcpy.